src/sgame/components/HealthComponent.cpp

```diff
@@ -119,11 +116,32 @@ Util::optional<Vec3> direction, int flags, meansOfDeath_t meansOfDeath) {
 
 	float take = amount;
 
+	// halve team damage according to the team, ignoring buildings
+	if (entity.oldEnt != source && G_OnSameTeam(entity.oldEnt, source)
+			&& entity.oldEnt->s.eType != entityType_t::ET_BUILDABLE) {
+		if ( G_Team(source) == TEAM_ALIENS )
+		{
+			take *= g_friendlyFireAlienMultiplier.value;
+		}
+		else
+		{
+			take *= g_friendlyFireHumanMultiplier.value;
+		}
+	}
+
+
 	// Apply damage modifiers.
 	if (!(flags & DAMAGE_PURE)) {
 		entity.ApplyDamageModifier(take, location, direction, flags, meansOfDeath);
 	}
 
+	// if we don't actually take damage, stop here
+	if ( take == 0.0f )
+	{
+		return;
+		// FIXME: this disables the damage indicator, but still shows blood
```

Comment:
Not a bad thing IMO, to show what you are hitting.

Reply:
Well, I would disagree, but I've removed the comment for now to avoid pushing that way. If someone gets annoyed by it they can discuss it.